standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java

```diff
@@ -679,7 +679,9 @@ public Partition alterPartition(RawStore msdb, Warehouse wh, String catName, Str
       // 1) partition SD Location
       // 2) partition column stats if there are any because of part_name field in HMS table PART_COL_STATS
       // 3) rename the partition directory if it is not an external table
-      if (!tbl.getTableType().equals(TableType.EXTERNAL_TABLE.toString())) {
+      boolean shouldMoveData = !(MetaStoreUtils.isExternalTable(tbl) &&
+          !MetaStoreUtils.isPropertyTrue(tbl.getParameters(), "TRANSLATED_TO_EXTERNAL"));
```

Comment:
Do we need to relocate the partitioned data of all the externally translated tables or only the ones that are translated with the config "hive.create.as.external.legacy"?

Reply:
only the ones thats are translated. We are trying to retain the legacy behavior. For tables created with "CREATE EXTERNAL", we assume that the partition's location is being written to or read from by external "processes". Does this make sense?